pkg/reconciler/utils/pubsub/topic.go

```diff
@@ -71,11 +71,13 @@ func (r *Reconciler) DeleteTopic(ctx context.Context, id string, obj runtime.Obj
 	exists, err := topic.Exists(ctx)
 	if err != nil {
 		logger.Error("Failed to verify Pub/Sub topic exists", zap.Error(err))
+		updater.MarkTopicUnknown("FinalizeTopicUnknown", "failed to verify Pub/Sub topic exists: %w", err)
 		return err
 	}
 	if exists {
 		if err := topic.Delete(ctx); err != nil {
 			logger.Error("Failed to delete Pub/Sub topic", zap.Error(err))
+			updater.MarkTopicFailed("FinalizeTopicFailed", "failed to delete Pub/Sub topic: %w", err)
```

Comment:
This is tricky. If we fail to delete the topic, depending on how you look at it, the Broker can be treated as "READY" because it can function properly.

Same comment for the pull subscription.

Reply:
Sync offline, we think marking unknown would be more reasonable than failed. If the topic deletion failed, you can't tell whether the remaining topic is still working or not. Probably the topic is still working after deletion failed, then, it is not proper to mark failed.